compiler/src/dotty/tools/dotc/typer/ConstFold.scala

```diff
@@ -28,10 +29,13 @@ object ConstFold {
               case _ => null
           case _ => null
       case Select(xt, op) =>
-        xt.tpe.widenTermRefExpr match {
-          case ConstantType(x) => foldUnop(op, x)
-          case _ => null
-        }
+        if (op eq nme.getClass_) && xt.tpe.widen.isPrimitiveValueType then
```

Comment:
Hum, this seems to handle `5.getClass` but not `5.getClass()` :s

Reply:
In Dotty:
```scala
scala> 5.getClass()
1 |5.getClass()
  |^^^^^^^^^^
  |expression does not take parameters
```

Reply:
Ouch. Is that the case for reference types as well? Or only for `Any` and `AnyVal`s? Because for reference types, `getClass` should be the Java-defined `java.lang.Object.getClass()` method, which should accept `()`.

Reply:
Ah yes, that works. in fact the reason `5.getClass()` also works on dotty master but not on this branch since `5.getClass` is seen first and replaced by a constant. I think that's acceptable.